src/components/common/Dialogs/CreatePaymentDialog/CreatePaymentDialogForm.tsx

```diff
@@ -0,0 +1,682 @@
+  //        * Because of this, we keep our own error state that runs in parallel
+  //        * to Formik's error state.
+  //        */
+  //       setCustomAmountError(MSG.noBalance);
+  //     } else {
+  //       setCustomAmountError(undefined);
+  //     }
+  //   }
+  // }, [
+  //   amount,
+  //   domainId,
+  //   fromDomainRoles,
+  //   fromDomainTokenBalance,
+  //   selectedToken,
+  //   setCustomAmountError,
+  //   networkFeeInverse,
+  // ]);
+
+  const userHasFundingPermission = userHasRole(
+    fromDomainRoles,
+    ColonyRole.Funding,
+  );
+  const userHasAdministrationPermission = userHasRole(
+    fromDomainRoles,
+    ColonyRole.Administration,
+  );
+  const hasRoles = userHasFundingPermission && userHasAdministrationPermission;
+  const requiredRoles: ColonyRole[] = [
+    ColonyRole.Funding,
+    ColonyRole.Administration,
+  ];
+
+  const [userHasPermission, onlyForceAction] = useDialogActionPermissions(
+    colony.colonyAddress,
+    hasRoles,
+    false, // isVotingExtensionEnabled,
+    values.forceAction,
+    domainId,
+  );
+
+  // const cannotCreateMotion =
+  //   votingExtensionVersion ===
+  //     VotingReputationExtensionVersion.FuchsiaLightweightSpaceship &&
+  //   !values.forceAction;
+
+  const handleFromDomainChange = useCallback(
+    (fromDomainValue) => {
+      const fromDomainId = parseInt(fromDomainValue, 10);
+      if (
+        fromDomainId !== Id.RootDomain &&
+        fromDomainId !== currentFromDomain
+      ) {
+        setCurrentFromDomain(fromDomainId);
+      } else {
+        setCurrentFromDomain(Id.RootDomain);
+      }
+      if (values.motionDomainId !== fromDomainId) {
+        setValue('motionDomainId', fromDomainId);
+      }
+    },
+    [currentFromDomain, setValue, values.motionDomainId],
+  );
+
+  // const handleFilterMotionDomains = useCallback(
+  //   (optionDomain) => {
+  //     const optionDomainId = parseInt(optionDomain.value, 10);
+  //     if (currentFromDomain === Id.RootDomain) {
+  //       return optionDomainId === Id.RootDomain;
+  //     }
+  //     return (
+  //       optionDomainId === currentFromDomain ||
+  //       optionDomainId === Id.RootDomain
+  //     );
+  //   },
+  //   [currentFromDomain],
+  // );
+
+  // const handleMotionDomainChange = useCallback(
+  //   (motionDomainId) => setFieldValue('motionDomainId', motionDomainId),
+  //   [setFieldValue],
+  // );
+
+  const canMakePayment = userHasPermission; // && isOneTxPaymentExtensionEnabled;
+
+  const inputDisabled = !canMakePayment || onlyForceAction || isSubmitting;
+
+  const formattingOptions = useMemo(
+    () => ({
+      delimiter: ',',
+      numeral: true,
+      numeralDecimalScale: getTokenDecimalsWithFallback(
+        selectedToken && selectedToken.decimals,
+      ),
+    }),
+    [selectedToken],
+  );
+
+  const formattedData = useMemo(
+    () => verifiedUsers.map((user) => ({ ...user, id: user.walletAddress })),
+    [verifiedUsers],
+  );
+
+  return (
+    <>
```

Comment:
Can you give me an example of something that could be encapsulated in this component? I feel like I understand what you are referring to but I want to make sure. For example:
```
        <div className={styles.tokenAmount}>
          <div className={styles.tokenAmountInputContainer}>
            <Input
              label={MSG.amount}
              name="amount"
              appearance={{
                theme: 'minimal',
                align: 'right',
              }}
              formattingOptions={formattingOptions}
              disabled={inputDisabled}
              dataTest="paymentAmountInput" 
            />
            {/* {networkFeeInverse &&
              customAmountError === undefined &&
              values.amount &&
              Number(values.amount) > 0 && (
                <div className={styles.networkFee}>
                  <FormattedMessage
                    {...MSG.fee}
                    values={{
                      fee: (
                        <Numeral
                          appearance={{
                            size: 'small',
                            theme: 'grey',
                          }}
                          value={
                            calculateFee(
                              values.amount,
                              networkFeeInverse,
                              getTokenDecimalsWithFallback(
                                selectedToken?.decimals,
                              ),
                            ).feesInWei
                          }
                          unit={getTokenDecimalsWithFallback(
                            selectedToken && selectedToken.decimals,
                          )}
                        />
                      ),
                      symbol: (selectedToken && selectedToken.symbol) || '???',
                    }}
                  />
                </div>
              )} */}
          </div>
          <div className={styles.tokenAmountContainer}>
            <div className={styles.tokenAmountSelect}>
              <TokenSymbolSelector
                label={MSG.token}
                tokens={colonyTokens}
                name="tokenAddress"
                elementOnly
                appearance={{ alignOptions: 'right', theme: 'grey' }}
                disabled={inputDisabled}
              />
            </div>
            {values.tokenAddress === AddressZero && (
              <div className={styles.tokenAmountUsd}>
                <EthUsd
                  // appearance={{ theme: 'grey' }}
                  value={amount}
                />
              </div>
            )}
          </div>
        </div>
```
Would you say that this part can be encapsulated further?

Reply:
I generally think of it like each distinct "thing" in the form is its own component, e.g. an input or a selector, which may itself be composed of smaller components. So in this form, I see: 

A Dialog Heading (which includes the domain selector and the title "Payment")
Another domain selector (from: Root)
Recipient selector
Amount selector (including the token selector)
The text area for the annotation
And the controls at the bottom (back / create motion)

I think that's what Jakub's suggesting also, because these are the things wrapped in the `DialogSection`s.

The nice thing about this exercise is that we are going to start to see patterns of repeated abstractions (every form has a heading with the domain selector and the controls at the bottom, for example), so it provides a basis for further code reuse. I'm basically using [this guide](https://reactjs.org/docs/thinking-in-react.html) as my reference when thinking about component hierarchies / encapsulation.

Reply:
To your question, I'd say this represent its own "thing" (e.g. NetworkFeeMessage): 

``` js
{/* {networkFeeInverse &&
              customAmountError === undefined &&
              values.amount &&
              Number(values.amount) > 0 && (
                <div className={styles.networkFee}>
                  <FormattedMessage
                    {...MSG.fee}
                    values={{
                      fee: (
                        <Numeral
                          appearance={{
                            size: 'small',
                            theme: 'grey',
                          }}
                          value={
                            calculateFee(
                              values.amount,
                              networkFeeInverse,
                              getTokenDecimalsWithFallback(
                                selectedToken?.decimals,
                              ),
                            ).feesInWei
                          }
                          unit={getTokenDecimalsWithFallback(
                            selectedToken && selectedToken.decimals,
                          )}
                        />
                      ),
                      symbol: (selectedToken && selectedToken.symbol) || '???',
                    }}
                  />
                </div>
              )} */}

```

This condition should really live in a named variable so its easier for others to see what it's trying to achieve: 

``` js
networkFeeInverse &&
              customAmountError === undefined &&
              values.amount &&
              Number(values.amount) > 0 
```

And then I would personally look at writing a function called `getValues` or `getMessage values` that returns the values object, so that the actual `FormattedMessage` component just looks something like:

``` js

const numeralValue = calculateFee(values.amount, networkFeeInverse, getTokenDecimalsWithFallback(selectedToken?.decimals)).feesInWei

...

<FormattedMessage
                    {...MSG.fee}
                    values={getValues(numeralValue, numeralUnit, tokenSymbol)}
```

And I think the following could probably just live inside a component called `TokenSymbol`

``` js
<div className={styles.tokenAmountContainer}>
            <div className={styles.tokenAmountSelect}>
              <TokenSymbolSelector
                label={MSG.token}
                tokens={colonyTokens}
                name="tokenAddress"
                elementOnly
                appearance={{ alignOptions: 'right', theme: 'grey' }}
                disabled={inputDisabled}
              />
            </div>
            {values.tokenAddress === AddressZero && (
              <div className={styles.tokenAmountUsd}>
                <EthUsd
                  // appearance={{ theme: 'grey' }}
                  value={amount}
                />
              </div>
            )}
          </div>

```

So the finished section might look like

``` js

<>
   <PaymentAmount>
  <NetworkFeeMessage>
  <TokenSymbol>
</>

```

I would add that in general, the intention is to make it easier to see what's happening at a glance, so it's easier for the next guy to understand what's going on, and this is often achieved through moving logic into named variables and functions (since the naming process itself is an exercise in signposting intent to team members).